cli/cmd/set.go

```diff
@@ -256,20 +254,25 @@ func loadWorkloadOwnerKey(path string, manifst manifest.Manifest, log *slog.Logg
 	if err != nil {
 		return nil, fmt.Errorf("parsing workload owner key: %w", err)
 	}
-	pubKey, err := x509.MarshalPKIXPublicKey(&workloadOwnerKey.PublicKey)
-	if err != nil {
-		return nil, fmt.Errorf("marshaling public key: %w", err)
-	}
-	ownerKeyHash := sha256.Sum256(pubKey)
-	ownerKeyHex := manifest.NewHexString(ownerKeyHash[:])
-	if len(manifst.WorkloadOwnerKeyDigests) == 0 {
-		log.Warn("No workload owner keys in manifest. Further manifest updates will be rejected by the coordinator")
-		return workloadOwnerKey, nil
-	}
-	log.Debug("Workload owner keys in manifest", "keys", manifst.WorkloadOwnerKeyDigests)
-	if !slices.Contains(manifst.WorkloadOwnerKeyDigests, ownerKeyHex) {
-		log.Warn("Workload owner key not found in manifest. This may lock you out from further updates")
+
+	// Check workload owner key configuration in manifest on set.
+	if manifst != nil {
+		pubKey, err := x509.MarshalPKIXPublicKey(&workloadOwnerKey.PublicKey)
+		if err != nil {
+			return nil, fmt.Errorf("marshaling public key: %w", err)
+		}
+		ownerKeyHash := sha256.Sum256(pubKey)
+		ownerKeyHex := manifest.NewHexString(ownerKeyHash[:])
+		if len(manifst.WorkloadOwnerKeyDigests) == 0 {
+			log.Warn("No workload owner keys in manifest. Further manifest updates will be rejected by the coordinator")
+			return workloadOwnerKey, nil
+		}
+		log.Debug("Workload owner keys in manifest", "keys", manifst.WorkloadOwnerKeyDigests)
+		if !slices.Contains(manifst.WorkloadOwnerKeyDigests, ownerKeyHex) {
+			log.Warn("Workload owner key not found in manifest. This may lock you out from further updates")
+		}
 	}
```

Comment:
Should this be part of the manifest validation? cc @davidweisse

Reply:
If this is possibly wanted behavior, then I don't think so. Also, the manifest validation only checks if all fields contain valid data and shouldn't require any extra keys for comparison. We should instead check if the seed share owner pub keys are valid public keys.